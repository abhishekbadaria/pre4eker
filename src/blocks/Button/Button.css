.button {
  padding: 5px 10px;
  border: 1px solid #000000;
}

.normal {
  background: #FFFFFF;
}

.hover {
  background: #27AE60;
}