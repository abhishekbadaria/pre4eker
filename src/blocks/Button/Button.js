import React, { useState } from "react";
import PropTypes from "prop-types";

import Shevron from "src/blocks/Icons/Shevron";
import Add from "src/blocks/Icons/Add";
import Remove from "src/blocks/Icons/Remove";
import style from "./Button.css";

export default function Button({ action, type, size, active, children }) {
  const typeOfButton = (buttonType) => {
    switch (buttonType) {
      case "DROPDOWN":
        return { icon: <Shevron />, styleClass: style.basic };
      case "ADD":
        return { icon: <Add />, styleClass: style.add };
      case "REMOVE":
        return { icon: <Remove />, styleClass: style.remove };
      case "TEXT":
        return { icon: children, styleClass: style.basic };

      default:
        return { icon: <Shevron />, styleClass: style.basic };
    }
  };

  const [status, changeStatus] = useState("normal");
  return (
    <button
      className={`${style[size]} ${style.button} ${style[status]} ${
        active && style.active
      } ${typeOfButton(type).styleClass}`}
      onClick={(e) => action(e)}
      onMouseEnter={() => changeStatus("hover")}
      onMouseLeave={() => changeStatus("normal")}
      type="button"
    >
      {typeOfButton(type).icon}
    </button>
  );
}

Button.defaultProps = {
  action: () => ({}),
  type: "",
  size: "medium",
  active: false,
  children: "",
};
Button.propTypes = {
  action: PropTypes.func,
  type: PropTypes.string,
  size: PropTypes.string,
  active: PropTypes.bool,
  children: PropTypes.string,
};
