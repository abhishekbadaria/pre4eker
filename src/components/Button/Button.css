.button {
  border: 1px solid #000000;
  display: block;
  justify-content: center;
  align-items: center;
  padding: 0;
  cursor: pointer;
  /* height: 0; */
}

.basic.normal,
.add.normal,
.remove.normal {
  background: #FFFFFF;
  color: #000000;
}

.basic.hover,
.basic.active {
  background: #000;
  color: #fff;
  border-color: #000;
}

.hover svg path {
  fill: white;
}

.add.hover {
  background: #27AE60;
  border-color: #27AE60;
  color: #fff;
}

.remove.hover {
  background: #EB5757;
  color: #fff;
  border-color: #EB5757;
}

.small {
  height: 14px;
  width: 14px;
}

.medium {
  height: 24px;
  width: 24px;
}

.sizeAuto {
  padding: .4rem .5rem .3rem .5rem;
}

.sizeFull {
  width: 100%;
  padding: .6rem .5rem .5rem .5rem;
}