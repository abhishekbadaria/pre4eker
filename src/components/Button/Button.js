import React, { useState } from "react";
import PropTypes from "prop-types";

import Shevron from "src/components/Icons/Shevron";
import Cross from "src/components/Icons/Cross";
import Add from "src/components/Icons/Add";
import Remove from "src/components/Icons/Remove";
import style from "./Button.css";

export default function Button({
  action,
  type,
  size,
  active,
  children,
  passedClassName,
}) {
  const typeOfButton = (buttonType) => {
    switch (buttonType) {
      case "DROPDOWN":
        return { icon: <Shevron />, styleClass: style.basic };
      case "ADD":
        return { icon: <Add />, styleClass: style.add };
      case "REMOVE":
        return { icon: <Remove />, styleClass: style.remove };
      case "TEXT":
        return { icon: children, styleClass: style.basic };
      case "CROSS":
        return { icon: <Cross />, styleClass: style.basic };

      default:
        return { icon: <Shevron />, styleClass: style.basic };
    }
  };

  const [status, changeStatus] = useState("normal");
  return (
    <button
      className={`${passedClassName} ${style[size]} ${style.button} ${
        style[status]
      } ${active && style.active} ${typeOfButton(type).styleClass}`}
      onClick={(e) => action(e)}
      onMouseEnter={() => changeStatus("hover")}
      onMouseLeave={() => changeStatus("normal")}
      type="button"
    >
      {typeOfButton(type).icon}
    </button>
  );
}

Button.defaultProps = {
  action: () => ({}),
  type: "",
  size: "medium",
  active: false,
  children: "",
  passedClassName: "",
};
Button.propTypes = {
  action: PropTypes.func,
  type: PropTypes.string,
  size: PropTypes.string,
  active: PropTypes.bool,
  children: PropTypes.string,
  passedClassName: PropTypes.string,
};
