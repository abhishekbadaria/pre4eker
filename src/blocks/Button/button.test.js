import React from "react";
import Button from "./Button";
import { render, fireEvent, waitFor, screen, cleanup, act } from '@testing-library/react'

afterEach(cleanup)


test(`Button changes color when hovered`, () => {
  const {asFragment, getByRole} = render(<Button type={`DROPDOWN`} action={() => null} />)
  
  expect(asFragment(<Button />)).toMatchSnapshot();
  
  
  fireEvent.mouseOver(screen.getByRole('button'))
  expect(asFragment(<Button />)).toMatchSnapshot();


  fireEvent.mouseLeave(screen.getByRole('button'))
  expect(asFragment(<Button />)).toMatchSnapshot();

});

test(`fire an action on click`, () => {
  const action = jest.fn();

  const {getByRole} = render(<Button type={`DROPDOWN`} action={action} />)

  fireEvent.click(screen.getByRole(`button`))
  expect(action).toHaveBeenCalled()
})