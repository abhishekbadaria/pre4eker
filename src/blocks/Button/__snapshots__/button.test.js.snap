// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button changes color when hovered 1`] = `
<DocumentFragment>
  <button
    class="medium button normal undefined basic"
  >
    <div>
      <svg
        fill="none"
        height="5"
        viewBox="0 0 8 5"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M0.0344238 1.16905L0.732916 0.502381L3.54531 3.1561L6.33428 0.499939L7.03473 1.16647L3.53314 4.50002L0.0344238 1.16905Z"
          fill="#000"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button changes color when hovered 2`] = `
<DocumentFragment>
  <button
    class="medium button hover undefined basic"
  >
    <div>
      <svg
        fill="none"
        height="5"
        viewBox="0 0 8 5"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M0.0344238 1.16905L0.732916 0.502381L3.54531 3.1561L6.33428 0.499939L7.03473 1.16647L3.53314 4.50002L0.0344238 1.16905Z"
          fill="#000"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button changes color when hovered 3`] = `
<DocumentFragment>
  <button
    class="medium button normal undefined basic"
  >
    <div>
      <svg
        fill="none"
        height="5"
        viewBox="0 0 8 5"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M0.0344238 1.16905L0.732916 0.502381L3.54531 3.1561L6.33428 0.499939L7.03473 1.16647L3.53314 4.50002L0.0344238 1.16905Z"
          fill="#000"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;
